File: bigjobhunterpro-web/src/types/timelineEvent.ts
export type EventType =
  | 'Prospecting'
  | 'Applied'
  | 'Screening'
  | 'Interview'
  | 'Offer'
  | 'Rejected'
  | 'Withdrawn';

export interface TimelineEvent {
  id: string;
  applicationId: string;
  eventType: EventType;
  interviewRound: number | null;
  timestamp: string;
  notes: string | null;
  points: number;
  createdDate: string;
}

export interface CreateTimelineEventRequest {
  eventType: EventType;
  interviewRound?: number;
  timestamp: string;
  notes?: string;
}

export interface UpdateTimelineEventRequest {
  eventType: EventType;
  interviewRound?: number;
  timestamp: string;
  notes?: string;
}

export interface TimelineEventsListResponse {
  applicationId: string;
  events: TimelineEvent[];
  totalPoints: number;
  currentStatus: string;
}

File: src/Domain/Entities/TimelineEvent.cs
using Domain.Enums;

namespace Domain.Entities;

public class TimelineEvent
{
    public Guid Id { get; set; }
    public Guid ApplicationId { get; set; }
    public EventType EventType { get; set; }
    public int? InterviewRound { get; set; }  // Only for Interview events
    public DateTime Timestamp { get; set; }
    public string? Notes { get; set; }
    public int Points { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime? UpdatedDate { get; set; }

    // Navigation property
    public Application Application { get; set; } = null!;
}
